lib/matplotlib/axes/_axes.py

```diff
@@ -8392,31 +8424,61 @@ def violin(self, vpstats, positions=None, vert=True, widths=0.5,
         # Calculate ranges for statistics lines (shape (2, N)).
         line_ends = [[-0.25], [0.25]] * np.array(widths) + positions
 
-        # Colors.
+        # Make a cycle of color to iterate through, using 'none' as fallback
+        def cycle_color(color):
```

Comment:
I do not understand this function, `itertools.cycle` repeats no we never hit the fallback to `none`.

Reply:
It is meant to copy the functionality in ``bar`` (see [here](https://github.com/matplotlib/matplotlib/blob/main/lib/matplotlib/axes/_axes.py#L2500)), in which ``"none"`` is a possible input to ``mcolors.to_rgba_array()`` that returns ``[]``. This is the specific case where the fallback to ``none`` is required.